contracts/swap/src/queries.rs

```diff
@@ -162,140 +234,166 @@ fn estimate_execution_buy(
 
     let funds_for_margin = match is_simulation {
         false => funds_in_contract, // / in execution mode funds_in_contract already contain user funds so we don't want to count them double
-        true => funds_in_contract + available_funds,
+        true => funds_in_contract + available_funds.unwrap_or_default(),
     };
-    deps.api.debug(&format!(
-        "estimate_execution_buy: required_funds: {}, funds_in_contract: {}, available_funds: {}: diff: {}, funds_for_margin: {}",
-        required_funds, funds_in_contract, available_funds, if !is_simulation { funds_in_contract - required_funds} else { funds_in_contract + available_funds - required_funds}, funds_for_margin
-    ));
 
     if required_funds > funds_for_margin {
-        Err(StdError::generic_err(format!(
+        return Err(StdError::generic_err(format!(
             "Swap amount too high, required funds: {required_funds}, available funds: {funds_for_margin}",
-        )))
-    } else {
-        Ok(StepExecutionEstimate {
-            worst_price,
-            result_quantity: round_to_min_tick(expected_quantity, market.min_quantity_tick_size),
-            result_denom: counter_denom(market, &amount_coin.denom)?.to_string(),
-            is_buy_order: true,
-            fee_estimate: Some(FPCoin {
-                denom: market.quote_denom.clone(),
-                amount: fee_estimate,
-            }),
-        })
+        )));
     }
+
+    Ok(StepExecutionEstimate {
+        worst_price,
+        result_quantity,
+        result_denom: counter_denom(market, &amount_coin.denom)?.to_string(),
+        is_buy_order: true,
+        fee_estimate: Some(FPCoin {
+            denom: market.quote_denom.clone(),
+            amount: fee_estimate,
+        }),
+    })
```

Comment:
Why not split buy estimate in two method, the same way as you did with sell? I think it would be easier to read than a series of ifs

Reply:
agreed, changed it.